packages/eth/src/aws.rs

```diff
@@ -0,0 +1,182 @@
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct AwsCredentialsProvider {
+    credentials: CredentialsProvider,
+}
+
+impl AwsCredentialsProvider {
+    #[cfg(feature = "test-helpers")]
+    pub fn new_static(access_key: impl Into<String>, secret_access_key: impl Into<String>) -> Self {
+        Self {
+            credentials: CredentialsProvider::Static(StaticProvider::new(
+                access_key.into(),
+                secret_access_key.into(),
+                None,
+                None,
+            )),
+        }
+    }
+
+    pub fn new_chain() -> Self {
+        Self {
+            credentials: CredentialsProvider::Chain {
+                environment_provider: EnvironmentProvider::default(),
+                profile_provider: ProfileProvider::new().ok(),
+                instance_metadata_provider: InstanceMetadataProvider::new(),
+                container_provider: ContainerProvider::new(),
+                web_identity_provider: WebIdentityProvider::from_k8s_env(),
+            },
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+enum CredentialsProvider {
+    #[cfg(feature = "test-helpers")]
+    Static(StaticProvider),
+    Chain {
+        environment_provider: EnvironmentProvider,
+        instance_metadata_provider: InstanceMetadataProvider,
+        container_provider: ContainerProvider,
+        profile_provider: Option<ProfileProvider>,
+        web_identity_provider: WebIdentityProvider,
+    },
+}
+
+#[async_trait::async_trait]
+impl ProvideAwsCredentials for AwsCredentialsProvider {
+    async fn credentials(&self) -> std::result::Result<AwsCredentials, CredentialsError> {
+        match &self.credentials {
+            #[cfg(feature = "test-helpers")]
+            CredentialsProvider::Static(provider) => provider.credentials().await,
+            CredentialsProvider::Chain {
+                environment_provider,
+                instance_metadata_provider,
+                container_provider,
+                profile_provider,
+                web_identity_provider,
+            } => {
+                // Copied from rusoto_core::credential::ChainProvider
+                if let Ok(creds) = environment_provider.credentials().await {
+                    return Ok(creds);
+                }
+                if let Some(ref profile_provider) = profile_provider {
+                    if let Ok(creds) = profile_provider.credentials().await {
+                        return Ok(creds);
+                    }
+                }
+                if let Ok(creds) = container_provider.credentials().await {
+                    return Ok(creds);
+                }
+                if let Ok(creds) = instance_metadata_provider.credentials().await {
+                    return Ok(creds);
+                }
+
+                // Added by us
+                if let Ok(creds) = web_identity_provider.credentials().await {
+                    return Ok(creds);
+                }
```

Comment:
To be in-line with normal AWS SDK behavior, this should go between `environment_provider` and `profile_provider` on line 158.